### PR TITLE
Indexversion

### PR DESCRIPTION
**Pull request by fabxc**
@Gouthamve @brian-brazil @grobie @Bplotka

Test is a bit hacked as it just overwrites the existing data. Should be fine for this intermediate test though – alternatively we can write th 100 LOC required to copy a dir in Go of course :P

### CODE REVIEW

File: repair.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/prometheus/tsdb/fileutil"
+)
+
+func repairBadIndexVersion(logger log.Logger, dir string) error {
+	// All blocks written by Prometheus 2.1 with a meta.json version of 2 are affected.
```
Comment (Bplotka) [inline repair.go]: worth to add exact revision when it happen?

**Comment by brancz** [comment]
lgtm

File: block.go [hunk 2]
```diff
@@ -195,14 +195,16 @@ func readMetaFile(dir string) (*BlockMeta, error) {
 	if err := json.Unmarshal(b, &m); err != nil {
 		return nil, err
 	}
-	if m.Version != 1 && m.Version != 2 {
```
[the remark this answers is not in the document]
Comment (fabxc) [reply block.go]: yes

**Comment by brancz** [comment]
Just discussed offline, that we'll try to make sure we put some sort of a fail safe mechanism in place to skip version 2 for the meta format, as if one would upgrade from 2.1.0 to the "next" meta version 2 then there may be possible conflicts. Therefore we should skip version 2 when we bump the meta version and go straight to 3 to avoid possible conflicts like that.

**Comment by fabxc** [comment]
All addressed.

**Pull request merged**